lib/clients/authenticated.js

```diff
@@ -3,8 +3,14 @@ const { signRequest } = require('../../lib/request_signer');
 const PublicClient = require('./public.js');
 
 class AuthenticatedClient extends PublicClient {
-  constructor(key, b64secret, passphrase, apiURI) {
-    super('', apiURI);
+  constructor(
+    key,
+    b64secret,
+    passphrase,
+    apiURI = 'https://api.gdax.com',
+    productID = null
```

Comment:
The intention behind this is to make sure that all calls that can change things have an explicit product set. Users shouldn't be trading BTC-USD because they forgot to include the product they actually wanted to trade in the constructor. This shouldn't be a concern for the public client and I would opt to leave it as it is for backwards compatibility.